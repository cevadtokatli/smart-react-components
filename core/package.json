{
  "name": "@smart-react-components/core",
  "version": "2.0.0-beta-0",
  "license": "MIT",
  "main": "index.js",
  "description": "SRC core includes the theme, Styled properties, and Element components.",
  "keywords": [
    "javascript",
    "typescript",
    "react",
    "styled-components"
  ],
  "author": {
    "name": "Cevad Tokatlı",
    "email": "cevadtokatli@hotmail.com"
  },
  "contributors": [
    {
      "name": "Cemil Tokatlı",
      "email": "cemil@tokatli.me"
    }
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/cevadtokatli/smart-react-components/tree/v2/core"
  },
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "yarn clear && tsc --outDir lib && yarn copy",
    "clear": "rm -rf lib",
    "copy": "cp package.json lib/package.json && cp README.md lib/README.md && cp ../LICENSE lib/LICENSE",
    "dev": "tsc -w --outDir lib",
    "lint": "eslint src",
    "release": "../scripts/release.sh core",
    "test": "jest"
  },
  "devDependencies": {
    "@types/color": "3.0.3",
    "@types/enzyme": "3.10.12",
    "@types/jest": "29.0.1",
    "@types/react": "18.0.19",
    "@types/react-dom": "18.0.6",
    "@types/styled-components": "5.1.26",
    "@wojtekmaj/enzyme-adapter-react-17": "0.6.7",
    "enzyme": "3.11.0",
    "eslint": "8.0.1",
    "jest": "29.0.3",
    "jsdom": "20.0.0",
    "jsdom-global": "3.0.2",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "styled-components": "5.3.5",
    "ts-jest": "29.0.0",
    "typescript": "4.8.3"
  },
  "dependencies": {
    "color": "4.2.3"
  },
  "peerDependencies": {
    "react": "^18.2.0",
    "styled-components": "^5.3.5"
  }
}
