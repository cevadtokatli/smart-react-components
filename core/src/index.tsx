export * from './theme'
export { default as createTheme } from './theme'
export * from './styled-props'
export * from './Element'
