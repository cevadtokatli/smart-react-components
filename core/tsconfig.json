{
  "extends": "../tsconfig.json",
  "include": ["src/**/*.tsx"],
  "exclude": ["node_modules", "src/**/*.test.tsx"]
}
