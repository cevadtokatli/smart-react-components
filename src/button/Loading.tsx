import React from "react"
import CSSTransition from "../transition/CSSTransition"
import Div from "../element/Div"

interface Props {
    element: JSX.Element
    status: boolean
    transitionClassName: string
    transitionType: string
    transitionDuration: number
    transitionShowAnimation: boolean
    transitionHideAnimation: boolean
}

const Loading: React.FC<Props> = ({element,status,transitionClassName,transitionType,transitionDuration,transitionShowAnimation,transitionHideAnimation}) => (
    <CSSTransition status={status} className={transitionClassName} type={transitionType} duration={transitionDuration} showAnimation={transitionShowAnimation} hideAnimation={transitionHideAnimation}>
        <Div position="absolute" zIndex="buttonLoading" left="0" top="0" width="100%" height="100%" display="flex" justifyContent="center" alignItems="center" data-src-loading="true">{element}</Div>
    </CSSTransition>
)

export default Loading