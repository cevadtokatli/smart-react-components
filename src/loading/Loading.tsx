import React from "react"
import styled from "styled-components"
import ElementProps from "../element/Props"
import {SizeProps} from "../props/size-props"
import Svg from "../element/Svg"

export interface Props extends ElementProps, SizeProps {
    type?: string
}

const customKeys = {
    size: (v, t) => `
        width: ${t.loading.size[v]};
        height: ${t.loading.size[v]};
    `
}

export default styled(Svg).attrs<Props>(({size="default",type="primary"}) => ({
    customKeys,
    children: <circle cx="22" cy="22" r="20" strokeWidth="4" />,
    viewBox: "0 0 44 44",
    size,
    type
}))<Props>(({theme,type}) => `
    animation: src-loading-spin-animation ${theme.src.loading.animation} infinite;

    circle {
        fill: none;
        stroke: ${theme.src.type[type].main};
        stroke-dasharray: 1,150;
        stroke-dashoffset: 0;
        stroke-linecap: round;
        animation: src-loading-spin-circle-animation ${theme.src.loading.animation} infinite;
    }

    @keyframes src-loading-spin-animation {
        from {
            transform: rotate(0deg);
        }
    
        to {
            transform: rotate(360deg);
        }
    }

    @keyframes src-loading-spin-circle-animation {
        0% {
            stroke-dasharray: 1,150;
            stroke-dashoffset: 0;
        }
    
        50% {
            stroke-dasharray: 90,150;
            stroke-dashoffset: -35;
        }
    
        100% {
            stroke-dasharray: 90,150;
            stroke-dashoffset: -124;
        }
    }
`)