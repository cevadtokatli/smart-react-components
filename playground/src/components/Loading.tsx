import { Div } from '@smart-react-components/core'
import { Loading } from '@smart-react-components/ui'
import React from 'react'

export const Color = () => (
  <Div display="flex" flexWrap="wrap">
      <Loading marginRight="$length.3" marginBottom="$length.3" />
      <Loading color="!secondary" marginRight="$length.3" marginBottom="$length.3" />
      <Loading color="!success" marginRight="$length.3" marginBottom="$length.3" />
      <Loading color="!danger" marginRight="$length.3" marginBottom="$length.3" />
      <Loading color="!warning" marginRight="$length.3" marginBottom="$length.3" />
      <Loading color="!info" marginRight="$length.3" marginBottom="$length.3" />
      <Loading color="!light" marginRight="$length.3" marginBottom="$length.3" />
      <Loading color="!dark" marginRight="$length.3" marginBottom="$length.3" />
  </Div>
)

export const Size = () => (
  <Div display="flex" flexWrap="wrap">
    <Loading size="small" marginRight="$length.3" marginBottom="$length.3" />
    <Loading marginRight="$length.3" marginBottom="$length.3" />
    <Loading size="large" marginRight="$length.3" marginBottom="$length.3" />
  </Div>
)
