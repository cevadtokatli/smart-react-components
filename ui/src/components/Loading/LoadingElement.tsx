import Circle from '@smart-react-components/core/Element/Circle'
import Svg from '@smart-react-components/core/Element/Svg'
import { StyledProps } from '@smart-react-components/core/styled-props'
import { ResponsiveProp, SizeProp } from '@smart-react-components/core/types'
import React from 'react'
import styled from 'styled-components'

interface Props extends
  StyledProps,
  ResponsiveProp<'loadingSize', SizeProp> {
  loadingColor: string
}

export default styled(Svg).attrs<Props>(({ loadingColor }) => ({
  children: <Circle cx="22" cy="22" r="20" stroke={loadingColor} strokeWidth={4} />,
  getLoadingSize: (v, t) => `
    height: ${t.$.size.loading[v]};
    width: ${t.$.size.loading[v]};
  `,
  viewBox: '0 0 44 44',
}))<Props>`
  animation: src-loading-animation 2000ms linear infinite;

  circle {
    animation: src-loading-circle-animation 2000ms linear infinite;
    fill: none;
    stroke-dasharray: 1,150;
    stroke-dashoffset: 0;
    stroke-linecap: round;
  }

  @keyframes src-loading-animation {
    from {
      transform: rotate(0deg);
    }

    to {
      transform: rotate(360deg);
    }
  }

  @keyframes src-loading-circle-animation {
    0% {
      stroke-dasharray: 1,150;
      stroke-dashoffset: 0;
    }

    50% {
      stroke-dasharray: 90,150;
      stroke-dashoffset: -35;
    }

    100% {
      stroke-dasharray: 90,150;
      stroke-dashoffset: -124;
    }
  }
`
