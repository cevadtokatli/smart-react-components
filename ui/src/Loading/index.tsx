import extractElementProps from '@smart-react-components/core/element-props'
import intrinsicStyledMarginProps, { IntrinsicStyledMarginProps } from '@smart-react-components/core/element-props/intrinsic-styled-margin-props'
import { SizeProps } from '@smart-react-components/core/types'
import React from 'react'
import LoadingElement from './LoadingElement'

export interface Props extends IntrinsicStyledMarginProps, SizeProps {
  color?: string
}

const Loading: React.FC<Props> = props => (
  <LoadingElement
    loadingColor={props.color}
    loadingSize={props.size}
    loadingSizeSm={props.sizeSm}
    loadingSizeMd={props.sizeMd}
    loadingSizeLg={props.sizeLg}
    loadingSizeXl={props.sizeXl}
    {...extractElementProps(props, [intrinsicStyledMarginProps])}
  />
)

Loading.defaultProps = {
  color: '$color.blue',
  size: 'medium',
}

export default Loading
