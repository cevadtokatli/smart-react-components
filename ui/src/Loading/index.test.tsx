import { wrapTheme } from '@smart-react-components/core/test'
import { render } from '@testing-library/react'
import React from 'react'
import Loading from './'

describe('<Loading />', () => {
  it('should render component', () => {
    const { asFragment } = render(wrapTheme(<Loading />))
    expect(asFragment()).toMatchSnapshot()
  })
})
