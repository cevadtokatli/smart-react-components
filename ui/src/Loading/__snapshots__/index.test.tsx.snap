// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> should render component 1`] = `
<DocumentFragment>
  .c2 {
  stroke: #0665d0;
  stroke-width: 4px;
}

.c0 {
  height: 20px;
  width: 20px;
}

.c1 {
  -webkit-animation: src-loading-animation 2000ms linear infinite;
  animation: src-loading-animation 2000ms linear infinite;
}

.c1 circle {
  -webkit-animation: src-loading-circle-animation 2000ms linear infinite;
  animation: src-loading-circle-animation 2000ms linear infinite;
  fill: none;
  stroke-dasharray: 1,150;
  stroke-dashoffset: 0;
  stroke-linecap: round;
}

<svg
    class="c0 c1"
    viewBox="0 0 44 44"
  >
    <circle
      class="c2"
      cx="22"
      cy="22"
      r="20"
    />
  </svg>
</DocumentFragment>
`;
